### PR TITLE
[typing]: Work on material module

### PR DESCRIPTION
**Pull request by Diogo-Rossi**
# Description

This PR adds literal constants to the `Material.py` module

# Backporting

This change should be backported to the previous releases.

- [x] 2016
- [x] 2017
- [x] 2018
- [x] 2019
- [x] 2020
- [x] 2021
- [x] 2022

## Backporting changes

The following changes need to be made when backporting:
- [ ] 2016-2021: Remove the `BILAMINA` type for the `Elastic` method
- [ ] 2016-2019: Remove the `ANAND, DARVEAUX, DOUBLE_POWER, POWER_LAW, and TIME_POWER_LAW` type for the `Creep` method.
- [ ] 2016-2019: Remove the `BINGHAM_PLASTIC and HERSCHEL-BULKLEY` type for the `GapFlow` method. 
- [ ] 2016-2019: Remove the `ANAND, DARVEAUX, DOUBLE_POWER, POWER_LAW, and TIME_POWER_LAW` type for the `Viscous` method.

# Type of change

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: src/abaqus/UtilityAndView/SymbolicConstant.py [hunk 1]
```diff
@@ -2646,3 +2646,15 @@ def __repr__(self) -> str:
...
+    DOUBLE_POWER = "DOUBLE_POWER"
+    TIME_POWER_LAW = "TIME_POWER_LAW"
+    BINGHAM_PLASTIC = "BINGHAM_PLASTIC"
+    HERSCHEL_BULKLEY = "HERSCHEL_BULKLEY"
+    BILAMINA = "BILAMINA"
+    REUSS = "REUSS"
+    VOIGT = "VOIGT"
+    INVERSED_MT = "INVERSED_MT"
+    EISO = "EISO"
+    PISO = "PISO"
```
Comment (haiiliin) [inline src/abaqus/UtilityAndView/SymbolicConstant.py:2660]: Should this be added to `abaqusConstants` module?

Comment (Diogo-Rossi) [reply src/abaqus/UtilityAndView/SymbolicConstant.py:2660]: Yeah, sorry

Comment (Diogo-Rossi) [reply src/abaqus/UtilityAndView/SymbolicConstant.py:2660]: Done on [1ce393c](https://github.com/haiiliin/abqpy/pull/2946/commits/1ce393cfefc7a88b13ffe5980a0babc4ab41dcbb)

**Comment by haiiliin** [comment]
I think another method to correct the `SymbolicConstant` is to write a script to infer from the docstrings automatically (for example, by using the use `inspect` module), which may be much easier.

**Comment by Diogo-Rossi** [comment]
> I think another method to correct the `SymbolicConstant` is to write a script to infer from the docstrings automatically (for example, by using the `inspect` module), which may be much easier.

Good idea. I don't know how to do it exactly, but maybe there is a easier way to process.

**Comment by haiiliin** [comment]
> > I think another method to correct the `SymbolicConstant` is to write a script to infer from the docstrings automatically (for example, by using the `inspect` module), which may be much easier.
> 
> Good idea. I don't know how to do it exactly, but maybe there is a easier way to process.

You may use `inspect` to get source lines of live objects, infer `SymbolicConstant` literals from docstrings by the `__doc__` attributes, and replace the signatures in the source lines.

**Review by haiiliin — APPROVED** [review]

**Comment by Diogo-Rossi** [comment]
> You may use `inspect` to get source lines of live objects, infer `SymbolicConstant` literals from docstrings by the `__doc__` attributes, and replace the signatures in the source lines.

Maybe [this package](https://pypi.org/project/docstring-parser/) could help.

**Comment by haiiliin** [comment]
> > You may use `inspect` to get source lines of live objects, infer `SymbolicConstant` literals from docstrings by the `__doc__` attributes, and replace the signatures in the source lines.
> 
> Maybe [this package](https://pypi.org/project/docstring-parser/) could help.

Nice!

**Pull request merged**